### DISCUSSION TITLE
Implement in-app camera

### DISCUSSION THREAD

**Discussion by mynksnh** [ente-io/ente#967] [Enhancements]
Add camera functionality to allow ente users to take pictures through the app, bypassing the device photo gallery.

The captured photos should only be accessible through the ente mobile app, any copies of the images stored locally on the device must be encrypted 

The flutter [camera](https://pub.dev/packages/camera) plugin can be used to implement the in-app camera. Detailed examples of the plugin can be found [here](https://github.com/flutter/plugins/blob/main/packages/camera/camera/example/lib/main.dart), in this [blog post](https://blog.logrocket.com/flutter-camera-plugin-deep-dive-with-examples/), and the associated [github repo](https://github.com/sbis04/flutter_camera_demo)

**Comment by mnvr** [comment]
Thank you for the helpful info! This is indeed on our roadmap, so the links you provide will help us a lot and save us a lot of research 🙏

**Comment by ethicalrevolution** [comment]
Brilliant. I am looking forward to this being rolled out as I feel it is an essential part of making ente useful for me (and anyone else who wants to ensure Apple and Google don’t get any access to our photos).

**Comment by ale-rinaldi** [comment]
Hello, just wanted to say I'd love this feature as well.

I'm currently using Stingle Photos and I'm looking for an alternative since its development seems to be stopped for years. This is an important feature for me, that Stingle has. It even has a separate icon for it in the launcher, that directly opens the camera and, unlike the main app, it won't require any authentication. This is feasible because photos are encrypted with the public key, which is always accessible. So you can just take out your phone, open the Camera and start taking photos. It will ask for fingerprint/password only if you try to access previous photos.

I'd love to see something like that in Ente, it's probably the only thing that retains me from migrating!
